src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/Handlers/DataSourceParser.cs

```diff
@@ -0,0 +1,77 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Data.SqlClient.SNI;
+using Microsoft.Data.SqlClientX.Handlers;
+
+namespace Microsoft.Data.SqlClient.Microsoft.Data.SqlClientX.Handlers
+{
+    /// <summary>
+    /// Handler to populate data source information required for transport handler
+    /// </summary>
+    internal class DataSourceParser : IHandler<ConnectionHandlerContext>
+    {
+        public IHandler<ConnectionHandlerContext> NextHandler { get; set; }
+
+        public async ValueTask Handle(ConnectionHandlerContext request, bool isAsync, CancellationToken ct)
+        {
+            SqlConnectionString result = new SqlConnectionString(request.connectionString);
```

Comment:
Move this object of the type `SqlConnectionString` into the `ConnectionHandlerContext` itself. 
Almost every handler will need to create it. So might as well put it in the Context. 
The existing code of SqlClient also uses a SqlConnectionString option for getting the connection string properties out.

Reply:
I wonder is this something that will be provided before starting the handler chain?

Reply:
Done, replaced the ConnectionString with SqlConnectionString